### PR TITLE
Added some more meat to the Java guidelines

### PR DESCRIPTION
**Pull request by cmiller-wikia**
Added sections on:

* Additional coding standards
* API Development
* Standard library dependencies

### CODE REVIEW

**Comment by pchojnacki** [comment]
Looks awesome.
+1 for avoiding checked exceptions
+1 for Guava, Gradle

Optional is pretty new for me (last time I did Java 1.6 was all the rage) but it sounds reasonable.

File: Java/README.md [hunk 1]
```diff
@@ -1,12 +1,50 @@
...
+
+In addition to other conventions, the following rules apply to APIs that are to be exposed to third party developers (i.e. through being open sourced).
+
+### No dependencies on third-party classes
+
+Public APIs should only expose classes or methods controlled by Wikia, or that are part of the Java standard library. Exposing third-party classes or methods in APIs will inevitably make it impossible to upgrade the third-party library without breaking the API.
+
+## Standard Libraries/Tools
+
+* *Logging:* [SLF4J](http://www.slf4j.org) is the standard logging API. Unless another is required by a services's framework/server, the slf4j-log4j bridge should be used as the back-end.
```
Comment (pchojnacki) [inline Java/README.md]: My vote would be for Logback as sl4j backend.
+ it has some nice logstash integration: https://github.com/logstash/logstash-logback-encoder

Comment (cmiller-wikia) [inline Java/README.md]: I've never used logback in anger, happy to go with consensus on that one.

Comment (cmiller-wikia) [inline Java/README.md]: I've changed it to just recommend against using java.util.logging for now :)

**Pull request merged**